src/game/utils/har_screencap.c

```diff
@@ -31,27 +31,51 @@ int har_screencaps_clone(har_screencaps *src, har_screencaps *dst) {
     return 1;
 }
 
-void har_screencaps_capture(har_screencaps *caps, object *obj, int id) {
-    if(caps->ok[id]) {
-        surface_free(&caps->cap[id]);
-        caps->ok[id] = false;
-    }
-
-    // Position
+static vec2i camera_position_for(object *obj) {
     vec2i size = object_get_size(obj);
     vec2i pos = object_get_pos(obj);
     int x_margin = (SCREENCAP_W - size.x) / 2;
     int y_margin = (SCREENCAP_H - size.y) / 2;
     int x_center = pos.x - size.x / 2;
     int y_center = (NATIVE_H - pos.y);
-    int x = clamp(x_center - x_margin, 0, NATIVE_W - SCREENCAP_W);
-    int y = clamp(y_center + y_margin, 0, NATIVE_H);
+
+    vec2i res;
+    res.x = clamp(x_center - x_margin, 0, NATIVE_W - SCREENCAP_W);
+    res.y = clamp(y_center + y_margin, 0, NATIVE_H);
+    return res;
+}
+
+void har_screencaps_capture(har_screencaps *caps, object *obj, object *obj2, int id) {
+    if(caps->ok[id]) {
+        surface_free(&caps->cap[id]);
+        caps->ok[id] = false;
+    }
+
+    game_state *gs = obj->gs;
+
+    // Position
+    vec2i pos = camera_position_for(obj);
+    vec2i size = {SCREENCAP_W, SCREENCAP_H};
+    if(obj2) {
+        // pan out when bots are apart
+        vec2i pos2 = camera_position_for(obj2);
+        size.x = abs(pos.x - pos2.x) + size.x;
+        size.y = SCREENCAP_H * size.x / SCREENCAP_W;
+        pos.x = min2(pos.x, pos2.x);
+        pos.y -= (size.y - SCREENCAP_H) / 2;
+    }
+
+    video_render_prepare();
+    gs->hide_ui = true;
+    game_state_render(gs);
+    gs->hide_ui = false;
+    video_render_finish_offscreen();
```

Comment:
This is missing half the blending work, since it's done in the RGBA step. You may need to add another render phase for the remapping work, and reduce RGBA part to just palette search step.

Reply:
The har_screencap (via video_area_capture) captures the indexed colors from the offscreen framebuffer

Reply:
No, it only captures the red channel with the current direct palette indexes. You will need to copy the whole RGBA color set for that. In that case though, the arguments in GBA channels may no longer match the required remapping data.

Reply:
This PR does not introduce this problem, nor attempt to solve it.
Should we block this PR on the lack of remapping support in video_area_capture?